### ISSUE TITLE
anno_barplot give each bar a color

### ISSUE THREAD

**Issue by mknafl** [jokergoo/ComplexHeatmap#409]
I am working with data where I have a bar plot and I would like the bar plot to be able to be colored differently using a function.  How can I do this?  The function would just be use one color for all bars that share a different data value or feature.  Any help is appreciated!  Thanks!
Mark

**Comment by jokergoo** [comment]
Have you checked following link? You can find various settings for barplot annotation.

https://jokergoo.github.io/ComplexHeatmap-reference/book/heatmap-annotations.html#barplot_annotation

**Comment by mknafl** [comment]
I figured it out.  `library(ComplexHeatmap)
library(circlize)
library(oce)
WaterfallPlusFeatures <- function(DataFilename,WaterfallVar,FeatureVars,PlotHeight,RowNameFontSize,HeatmapAsFile)
{
	dRead<-read.csv(DataFilename)
	dSorted <- as.data.frame(dRead)
	WaterfallColumn <- grep(WaterfallVar,colnames(dSorted))
	dSorted <- dSorted[with(dSorted,order(-dSorted[,WaterfallColumn])),]
	dSmaller <- as.data.frame(t(dSorted[FeatureVars]))
	dSmaller[dSmaller==1.79769e+308]<-NA
	d <- as.matrix(dSmaller)
	dRespID <- as.matrix(dSorted[c("JustSlideID")])
	uRespID <- unique(dRespID)
	MyColorIndices <- data.frame(MyIndex = numeric(length(dRespID)))
	for(Index in seq(1,length(dRespID),by = 1))
	{
	  MyColorIndices$MyIndex[Index] <- grep(dRespID[Index],uRespID)
	}
	dResponse <- as.data.frame(dSorted[c(WaterfallVar)])
	BarChart = anno_barplot(dResponse, height = unit(PlotHeight, "cm"), gp = gpar(fill = MyColorIndices$MyIndex))`
Basically, I saved each bars grouping var, "JustSlideID", and then assigned each the appropriate unique("JustSlideID") index, and then used that as the vector for fill to choose a color from.
Thanks for making such a useful class!  Hopefully this helps someone doing something similar.
